deploy/cephcsi/image/Dockerfile

```diff
@@ -1,15 +1,50 @@
+ARG SRC_DIR="/go/src/github.com/ceph/ceph-csi/"
+
 FROM ceph/ceph:v15
-LABEL maintainers="Ceph-CSI Authors"
-LABEL description="Ceph-CSI Plugin"
 
-# To support cross building, do NOT RUN native binaries here.
-# If we have to run native binaries, qemu-user-static must be installed on
-# build host and mounted to container.
+LABEL stage="build"
+
+ARG GOLANG_VERSION=1.13.9
+ARG CSI_IMAGE_NAME=quay.io/cephcsi/cephcsi
+ARG CSI_IMAGE_VERSION=canary
+ARG ARCH
+ARG GIT_COMMIT
+
+RUN dnf install libcephfs-devel librados-devel librbd-devel /usr/bin/cc /usr/bin/c++ make -y
+
+RUN mkdir -p /usr/local/go && \
+    curl https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz | tar xzf - -C /usr/local/go --strip-components=1
+
+
+ENV GOROOT=/usr/local/go \
+    GOPATH=/go \
+    PATH=$GOROOT/bin:$GOPATH/bin:$PATH\
```

Comment:
This will not work on a single line. `${GOPROOT}` is not set in a previous command, so it will be empty here. Set is as an `ARG` before, and you can use it everywhere, including the Golang installation above.